cdap-app-fabric/src/test/java/co/cask/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -283,13 +283,18 @@ public Boolean call() throws Exception {
       }
     }, 10, TimeUnit.SECONDS);
 
-    // There shouldn't be any partition trigger in the job queue
-    Assert.assertFalse(Iterables.any(getAllJobs(), new Predicate<Job>() {
+    // Add timeout to allow jobs to get deleted on time
```

Comment:
This test became more flaky after these changes. @maochf and I were thinking that the `ConstraintCheckerService` became a bit slower in checking and updating job statuses.

@chtyim What do you think is the best approach to fix this? Rather than adding a timeout, I was thinking this check doesn't have to occur at all, since verifying that the partition schedules did not trigger might be a better idea than checking the job queue itself.